### ISSUE TITLE
Contest rating changing without giving the contest

### ISSUE THREAD

**Issue by HarshithBeta** [Being-Zero/mentorpick-issues#36]
source : ( I haven't participated in contest , but my rating is updated , and **it happened many times**)

![Screenshot (656)](https://github.com/user-attachments/assets/4eb072ed-bcb3-4df4-beb5-c52f8069e781)

![Screenshot (657)](https://github.com/user-attachments/assets/b72ee7e3-ea27-4507-91bf-9e348fd94a57)

Generally when i don't give "speed and accuracy" contest , i will open the contest after it ends and check the questions , i.e i request access to contest after the contest ends , but in this case my contest rating is getting decreased because the server is considering that i attempted the contest .
In short , if i request access to a contest after it is over , it shouldn't show my name in the leaderboard , but now it is showing due to which my rating is changing negatively.

SUGGESTION : only users who request access during the contest should have their name in leaderboard and only those people's rating should be updated , users who registered after the contest is completed shouldn't be affected.

**Comment by vishal-beingzero** [comment]
Thanks for letting us know, This issue is being investigated, we will let you know via this issue thread what action/precaution will be taken.

**Comment by vishal-beingzero** [comment]
This was for contest #137, We cannot fix the past ratings, but going forward we factor in the registration time. As long as user registers exactly after 10 PM (not even one millisecond before that), they will be excluded in the ratings.

**Comment by mitesh-bz** [comment]
Thank you for reporting this bug @HarshithBeta 
This has been fixed and we no longer include users registering for the contest after it ends in the rating computation. This fix was deployed in 3rd week of November. Unfortunately we cannot recompute the past ratings, so they will remain as is.
